### PR TITLE
Update gcenv.object.h

### PR DESCRIPTION
**Pull request by wayne-o**
Bringing it back to street level

### CODE REVIEW

File: src/gc/env/gcenv.object.h [hunk 1]
```diff
@@ -120,7 +120,7 @@ class MethodTable
 
     bool SanityCheck()
     {
-        return true;
+        return false;
```
Comment (jkotas) [inline src/gc/env/gcenv.object.h]: Could you please explain the motivation for this change? Why would we want to fail the sanity check unconditionally?

Comment (lkinsella) [reply src/gc/env/gcenv.object.h]: Why even have it at all if all it does is return true or false?

Comment (jkotas) [reply src/gc/env/gcenv.object.h]: The GC is build in several different configurations. Other configurations provide non-empty implementation of this method: https://github.com/dotnet/coreclr/blob/37b9d85941c39cfdce2a2ea877388ab1ab630c68/src/vm/methodtable.cpp#L8037

Comment (lkinsella) [reply src/gc/env/gcenv.object.h]: Ahh I see, was curious as seemed a pointless method otherwise!

**Pull request closed without merging**